### PR TITLE
Documentation: Add troubleshooting section to L2 Announcements

### PR DESCRIPTION
**Pull request by dylandreimerink**
Since its introduction, there has been a small yet steady stream of similar issues reported by users who are trying to use L2. This PR adds a troubleshooting section to the L2 Announcements which should help users to diagnose and resolve common issues on their own.

### CODE REVIEW

File: Documentation/network/l2-announcements.rst [hunk 1]
```diff
@@ -382,6 +390,180 @@ has been reached.
...
+
+Next, ensure you have at least one policy configured, L2 announcements will not work without a policy.
+
+.. code-block:: shell-session
+
+    $ kubectl get CiliumL2AnnouncementPolicy
+    NAME      AGE
+    policy1   6m16s
+
+L2 announcements should no create a lease for very service matched by the policy. We can check the leases like so:
```
Comment (ghost) [inline Documentation/network/l2-announcements.rst]:
```suggestion
L2 announcements should not create a lease for very service matched by the policy. We can check the leases like so:
```

File: Documentation/network/l2-announcements.rst [hunk 2]
```diff
@@ -382,6 +390,180 @@ has been reached.
...
+Another common error is that the configured client rate limit is too low. 
+This can be seen in the logs as well:
+
+.. code-block:: shell-session
+
+    $ kubectl -n kube-system logs ds/cilium | grep "l2"
+    2023-07-04T14:59:51.959400310Z level=info msg="Waited for 1.395439596s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:6443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/cilium-l2announce-default-example" subsys=klog
+    2023-07-04T15:00:12.159409007Z level=info msg="Waited for 1.398748976s due to client-side throttling, not priority and fairness, request: PUT:https://127.0.0.1:6443/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/cilium-l2announce-default-example" subsys=klog
+
+These logs are associated intermittent failures to renew the lease, connection issues and/or frequent leader changes.
```
Comment (ghost) [inline Documentation/network/l2-announcements.rst]:
```suggestion
These logs are associated with intermittent failures to renew the lease, connection issues and/or frequent leader changes.
```

File: Documentation/network/l2-announcements.rst [hunk 3]
```diff
@@ -382,6 +390,180 @@ has been reached.
...
+    lxc_health        16      false      veth     1500    0a:94:bf:49:d5:50   up|broadcast|multicast   fe80::894:bfff:fe49:d550
+    cilium_host       3       false      veth     1500    22:32:e2:80:21:34   up|broadcast|multicast   10.244.1.239, fd00:10:244:1::f58a
+
+Only devices with ``Selected`` set to ``true`` can be used for L2 announcements. Typically all physical devices with IPs
+assigned to them will be considered selected. The ``--devices`` flag or ``devices`` Helm option can be used to filter
+out devices. If your desired device is in the list but not selected, check the devices flag/option to see if it filters it out.
+
+Please open a Github issue if your desired device doesn't appear in the list or it isn't selected while you believe it should be.
+
+If the L2 state contains the IP and device combination but there are still connection issues, its time to test ARP
```
Comment (ghost) [inline Documentation/network/l2-announcements.rst]:
```suggestion
If the L2 state contains the IP and device combination but there are still connection issues, it's time to test ARP 
```

File: Documentation/network/l2-announcements.rst [hunk 4]
```diff
@@ -382,6 +390,180 @@ has been reached.
...
+It is still possible that the service is unreachable even though ARP requests are answered. This can happen 
+for a number of reasons, usually unrelated to L2 announcements, but rather other Cilium features.
+
+One common issue however is caused by the usage of ``.Spec.ExternalTrafficPolicy: Local`` on services. This setting
+normally tells a load balancer to only forward traffic to nodes with at least 1 ready pod to avoid a second hop.
+Unfortunately, L2 announcements isn't currently aware of this setting and will announce the service IP on all nodes
+matching policies. If a node without a pod receives traffic, it will drop it. To fix this, set the policy to 
+``.Spec.ExternalTrafficPolicy: Cluster``.
+
+Please make a Github issue if non of the above steps helped you solve your issue.
```
Comment (ghost) [inline Documentation/network/l2-announcements.rst]:
```suggestion
Please open a Github issue if none of the above steps helped you solve your issue.
```

File: Documentation/network/l2-announcements.rst [hunk 5]
```diff
@@ -382,6 +390,180 @@ has been reached.
...
+            },
+            "value": {
+                "responses_sent": 20
+            }
+        }
+    ]
+
+The ``responses_sent`` field is incremented every time the datapath responds to an ARP request. If the field
+is 0, then the ARP request doesn't make it to the node. If the field is greater than 0, the issue is on the
+return path. In both cases, inspect the network and or client.
```
Comment (ghost) [inline Documentation/network/l2-announcements.rst]:
```suggestion
return path. In both cases, inspect the network and the client.
```

**Review by ghost — APPROVED** [review]
Left a few grammar nitpicking comments, but LGTM.

Is it fine to backport these changes to v1.15 and v1.14 or will the steps not work for older releases?

**Comment by dylandreimerink** [comment]
Thanks for the review!

> Is it fine to backport these changes to v1.15 and v1.14 or will the steps not work for older releases?

Not all of the steps. Specifically the statedb commands will not work, so that would require some modification. I will do those manually if I find some time.

**Pull request merged**